avm/res/service-fabric/cluster/main.bicep

```diff
@@ -29,8 +29,8 @@ param maxUnusedVersionsToKeep int = 3
 @description('Optional. The settings to enable AAD authentication on the cluster.')
 param azureActiveDirectory object = {}
 
-@description('Optional. Describes the certificate details like thumbprint of the primary certificate, thumbprint of the secondary certificate and the local certificate store location.')
-param certificate object = {}
+@description('Required. Describes the certificate details like thumbprint of the primary certificate, thumbprint of the secondary certificate and the local certificate store location.')
```

Comment:
This is a case where I'd suggest to improve the usage with User Defined Types. The object does not seem a complex one (only contains 3 properties) but one is required, one is optional and one has restricted allowed values. It would improve the UX a lot to add a UDT definition for the certificate parameter.

Reply:
Agreed, added the certificateType UDT.

Reply:
This looks a perfect scenario for the Conditional parameter type 😉 Ref https://azure.github.io/Azure-Verified-Modules/specs/bicep/#id-bcpnfr7---category-inputs---parameter-requirement-types

Both parameters should then be of type `Conditional` and should contain a sentence "Required if <condition>" to explain the condition which would make them Required. In this case the condition should be that the other parameter is not specified.

Reply:
Ah yes, thanks for pointing that out. Updated both parameters to be conditional. Also added UDT's for certificateCommonNames, clientCertificateCommonNames, and clientCertificateThumbprints.